File: solutions/javascript/leap/1/leap.js

function isEvenlyDivisible(n, m) {
  return (n % m) == 0;
}

// Checks if the given year is a leap year.
export function isLeap(year) {
  
  // A year is a leap year if
  // - it is evenly divisible by 4
  // - it is not evenly divisible by 100 unless it is also evenly divisible by 400
  return isEvenlyDivisible(year, 4) && (!isEvenlyDivisible(year, 100) || isEvenlyDivisible(year, 400));
};

File: solutions/go/leap/1/leap.go
//Package leap provides a function to check if a year is a leap year.
package leap

// IsLeapYear check if a yer is  leap year.
func IsLeapYear(year int) bool {
	return (year % 4 == 0) && (year % 100 != 0 || year % 400 == 0)
}
